terminalkurs/TERMINALKURSPLAN.MD

```diff
@@ -0,0 +1,62 @@
+
+# TD-terminal:
+
+## Problemet: selge det inn til studentene
+
+### usecases:
+    * Lag flere lag av mapper i en enkel kommando
+    * Reproduserbare handlinger og eksperimenter
+    * Sette sammen kjøring av flere programmer(pipes, scripts, makefiles)
+    * Senere på uit og i arbeidslivet MÅ dere bruke terminal; innlogging på servere via ssh etc.
+
+Nyttig krydder:
+    * pushd (push directory) og popd (pop directory)
+
+    * ncal -w : se hvilken dato det er og hvilken uke man er i
+
+
+
+
+### Ønsket end state hos studenter etter å ha tatt kurs og øvet litt på oppgaver
+
+#### Ferdigheter
+* Studenten kan navigere i terminalen ved hjelp av cd, ls, pwd, pushd, popd
+* Studenten kan lage nye filer og mapper ved hjelp av touch og mkdir
+* Studenten kan lage nøstede mappestrukturer ved hjelp av mkdir -p
+* Studenten kan flytte filer og gi filer nytt navn ved hjelp av mv
+* Studenten kan kopiere filer ved hjelp av cp
+* Studenten kan slette hele mapper ved hjelp av rm -rf og tomme mapper ved hjelp av rmdir
+
+* Studenten kan kombinere kommandoer i terminalen ved hjelp av &&
+* Studenten kan binde sammen output fra ett program til input til et annet ved hjelp av pipes |
+* Studenten kan bruke cat, head og tail til inspisere filer
+* Studenten kan bruke redirects (> og >>) for å skrive standard output til fil.
+
+* Kunne kjøre programmer i terminalen (executables og f.eks python-programmer)
+
+#### Kunnskap
+* Studenten kan gjøre rede for filsystemets trestruktur
+* Studenten kjenner til at det finnes ulike måter å kommunisere med maskinen i en terminal ved
+  hjelp av shell-språk som f.eks bash, zsh og fish
+* Kjenne til verktøy som Ipython
+
+
+
+#### Kompetanse
+* Studenten kan forstå enkle shell-scripts
+
+
+
+#### Intermediate ferdigheter og krydder
```

Comment:
Her e noen ekstra som kanskje kan taes med om ønskelig
## Avanserte Læringsmål:

### Systemovervåking og -forvaltning:
- Overvåke systemressurser og prosesser med `top`, `htop`, `ps`, `netstat`.
- Forstå jobbkontroll med `bg`, `fg`, `jobs`, og signaler som `kill`.

### Programvareinstallasjon og -forvaltning:
- Installere, oppdatere og fjerne programvare ved hjelp av pakkesystemet spesifikt for distribusjonen (som `apt`, `yum`, `pacman`).

### Diskhåndtering:
- Forstå og bruke kommandoer for diskhåndtering som `df`, `du`, `mount`, `umount`, `fdisk`.

### Sikkerhetskopiering og Gjenoppretting:
- Utføre enkel sikkerhetskopiering og gjenoppretting av data med kommandoer som `tar`, `rsync`.